File: backend/elements/compute/src/lambda.ts
import { log, logJson } from '@guided/logger'
import action from './action'

export async function handler(_event: any, _context: any) {
    log('handler!')
    logJson(_event, '_event')
    logJson(_context, '_context')
    const computationId = _event['Records'][0]['body']
    logJson(computationId, 'computationId')
    await action({
        computationId,
    })
}

File: go/amend_dates/lambda.go
package main

import (
	"context"
	"fmt"
	"github.com/aws/aws-lambda-go/events"
	"github.com/aws/aws-lambda-go/lambda"
	"guided/utils"
)

func handler(_ context.Context, event events.SQSEvent) (string, error) {

	for _, message := range event.Records {
		fmt.Printf("The message %s for event source %s = %s \n", message.MessageId, message.EventSource, message.Body)
	}

	if len(event.Records) != 1 {
		err := utils.NewError("Expected exactly 1 message")
		fmt.Println(err)
		return "", err
	}

	guideId := event.Records[0].Body

	return Invoke(guideId)
}

func Lambda() {
	lambda.Start(handler)
}
